### PR TITLE
Fix empty features parsing

### PR DESCRIPTION
**Pull request by ryanbrainard**
If a user does not supply the `--features` flag, an empty string in an array (i.e.`[""]`) is being sent to API and fails with `Private Space features cannot be selected`. This handles parsing the null/empty case for features.

Note, I extracted out a `splitCsv()` function for this for better testing and reusablity. This should maybe belong the common CLI utils and/or the base CLI app as an option on the `flags`. Adding it here for now to fix the bug, but we can explore that later. cc: @dickeyxxx for guidance

cc: @ricardochimal, @jingweno 

https://www.pivotaltracker.com/story/show/129579099

### CODE REVIEW

File: commands/create.js [hunk 1]
```diff
@@ -14,7 +15,7 @@ function * run (context, heroku) {
       organization: context.org,
       channel_name: context.flags.channel,
       region: context.flags.region,
-      features: (context.flags.features || '').split(',').map((s) => { return s.trim() }),
```
Comment (dickeyxxx) [inline commands/create.js]: you could alternatively do this:

```js
features: context.flags.features ? context.flags.features.split(',').map(s => s.trim()) : null,
```

Comment (dickeyxxx) [inline commands/create.js]: but there are plans to have parsing and defaulting of flags at some point, but it would require a pure-node CLI

Comment (ryanbrainard) [inline commands/create.js]: that would fix the bug at hand, but that doesn't work for something like `'a,,c'`. Using `filter()` works for both cases and is a little cleaner/functional IMHO.

**Comment by jingweno** [comment]
Good find! LGTM.

**Comment by ryanbrainard** [comment]
I overrode the Changeling multipass because it didn't pick up the `Good find! LGTM.` correctly.

**Pull request merged**